test/performance/search/index_construction_benchmark.cpp

```diff
@@ -0,0 +1,132 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE
+// -----------------------------------------------------------------------------------------------------
+
+// written for the purpose of the Bch Softwarepraktikum, early 2019 FU Berlin
+// @author: Clemens Cords <clemenscords@fu-berlin.de>
+
+#include <cstring>
+#include <random>
+
+#include <benchmark/benchmark.h>
+
+#include <seqan3/io/stream/debug_stream.hpp>
+#include <seqan3/search/fm_index/all.hpp>
+#include <seqan3/alphabet/all.hpp>
+
+using namespace seqan3;
+
+const std::size_t LENGTH = 50;
+const unsigned int SEED = 1234;
+
+// ============================================================================
+// generate sequence deterministically based on fixed const seed
+// ============================================================================
+template<Alphabet alphabet_t>
+std::vector<alphabet_t> generate_test_sequence(const std::size_t length)
```

Comment:
Nice work on this function! I'm sorry though to tell you that we already have a function like this in `seqan3/test/include/seeqan3/test/performance/sequence_generator.hpp` which looks very much like yours but is already in the code base so it would be great if you use it.
It also just returns a vector over alphabet_t.